src/nvim/option.c

```diff
@@ -2568,6 +2568,69 @@ static bool valid_spellfile(const char_u *val)
   return true;
 }
 
+/// Handle setting 'mousescroll'.
+/// @return error message, NULL if it's OK.
+static char_u *check_mousescroll(char_u *string)
+{
+  long vertical = -1;
+  long horizontal = -1;
+
+  for (;;) {
+    char_u *end = vim_strchr(string, ',');
+    size_t length = end ? (size_t)(end - string) : STRLEN(string);
+
+    // Both "ver:" and "hor:" are 4 bytes long.
+    // They should be followed by at least one digit.
+    if (length <= 4) {
+      return e_invarg;
+    }
+
+    long *direction;
+
+    if (memcmp(string, "ver:", 4) == 0) {
+      direction = &vertical;
+    } else if (memcmp(string, "hor:", 4) == 0) {
+      direction = &horizontal;
+    } else {
+      return e_invarg;
+    }
+
+    // If the direction has already been set, this is a duplicate.
+    if (*direction != -1) {
+      return e_invarg;
+    }
+
+    // Verify that only digits follow the colon.
+    for (size_t i = 4; i < length; i++) {
+      if (!ascii_isdigit(string[i])) {
+        return (char_u *)N_("E548: digit expected");
+      }
+    }
+
+    errno = 0;
+    *direction = strtol((char *)string + 4, NULL, 10);
```

Comment:
There's a family of `getdigits` functions in src/nvim/charset.c that encapsulate this sort of functionality and error-checking.  Probably better to use the appropriate one of those.

Reply:
I had a look, I think the best way to do it is something like this:
```
string += 4;
*direction = getdigits_int(&string, false, -1);

// Num options are generally kept within the signed int range.
// We know this number won't be negative because we've already checked for
// a minus sign. We'll allow 0 as a means of disabling mouse scrolling.
if (*direction == -1) {
  return e_invarg;
}
```
Idk if that's helping readability, but it's your call.